vic/drivers/classic/src/read_initial_model_state.c

```diff
@@ -688,4 +692,90 @@ read_initial_model_state(FILE            *init_state,
             }
         }
     }
+
+    /******Check that soil moisture does not exceed maximum allowed************/
```

Comment:
Well you said that the `vic_populate_model_state()` is analogous to the `image` driver side.  I'm wondering if we want to pull out the common pieces? I'm trying to do a side by side comparison of the two implementations to assess that possibility.